_posts/2019-07-30-censored-data/censored_data_tfp.Rmd

```diff
@@ -0,0 +1,724 @@
+
+Why do we care about the censored portion, the entities for whom the event of interest did not occur in the period of observation? In the cat adoption scenario, this is pretty obvious: We want to be able to get a realistic estimate for any unknown cat, not just those cats that will turn out to be "lucky". How about `check_times`? Well, if your submission is one of those that errored out, you still care about how long you wait, so even though their percentage is low (< 1%) we don't want to simply exclude them. Also, there is the possibility that the failing ones would have taken longer, had they run to completion, dut to some intrinsic difference between both groups. Conversely, if failures were random, the longer-running checks would have a greater chance to get hit by an error. So here too, exluding the censored data may result in bias.
+
+How can we model durations for that censored portion, where the "true duration" is unknown? Taking one step back, how can we model durations in general? Making as few assumptions as possible, the [maximum entropy distribution](https://en.wikipedia.org/wiki/Maximum_entropy_probability_distribution) for displacements (in space or time) is the exponential. Thus, for the checks that actually did complete, durations are assumed to be exponentially distributed. 
+
+For the others, all we know is that in a virtual world where the check completed, it would take _at least as long_ as the given duration. This quantity can be modeled by the exponential complementary cumulative distribution function (CCDF). Why? A cumulative distribution function (CDF) indicates the probability that a value lower or equal to some reference point was reached; e.g., "the probability of durations <= 255 is 0.9". Its complement, 1 - CDF, then gives the probability that a value will exceed than that reference point.
+
+Let's see this in action.
+
+## The data
+
+The following code works with the current stable releases of TensorFlow and TensorFlow Probability, which are 1.14 and 0.7, respectively. If you don't have `tfprobability`  installed, get it from Github:
+
+```{r}
+remotes::install_github("rstudio/tfprobability")
+```
+
+These are the libraries we need. As of TensorFlow 1.14, we call `tf$compat$v2$enable_v2_behavior()` to run with eager execution.
+
+```{r}
+library(tensorflow)
+library(tfprobability)
+library(parsnip)
+library(tidyverse)
+library(zeallot)
+library(gridExtra)
+library(HDInterval)
+
+tf$compat$v2$enable_v2_behavior()
+```
+
+Besides the check durations we want to model, `check_times` reports various features of the package in question, such as number of imported packages, number of dependencies, size of code and documentation files, etc. The `status` variable indicates whether the check completed or errored out.
```

Comment:
```suggestion
Besides the check durations we want to model, `check_times` reports' various features of the package in question, such as number of imported packages, number of dependencies, size of code and documentation files, etc. The `status` variable indicates whether the check completed or errored out.
```

Reply:
I think this is a misunderstanding - "reports" is supposed to be read as a verb - had I better choose a different one?

Reply:
(>ლ) No, that makes total sense — I think I wasn't reading in "code mode"!

Reply:
okay thanks!